bigtable-hbase-parent/bigtable-hbase/src/main/java/org/apache/hadoop/hbase/client/AbstractBigtableConnection.java

```diff
@@ -73,6 +74,11 @@ public void run() {
       }
     };
     Runtime.getRuntime().addShutdownHook(new Thread(shutDownRunnable));
+
+    // Force the loading of HConstants.class, which shares a bi-directional reference with KeyValue.
+    // This bi-drectional relationship causes problems when KeyVale and HConstants are class loaded
+    // in different threads.  This forces a clean class loading of both classes.
+    HConstants.class.getName();
```

Comment:
The problem is a bidirectional relationship between `KeyValue` and `HConstant`.  Is there a similar bidirectional relationship with `Adapters`?  What does loading `Adapters` achieve?